src/main/java/com/redhat/devtools/lsp4ij/features/formatting/LSPOnTypeFormattingSupport.java

```diff
@@ -0,0 +1,97 @@
+/*******************************************************************************
+ * Copyright (c) 2025 Red Hat, Inc.
+ * Distributed under license by Red Hat, Inc. All rights reserved.
+ * This program is made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution,
+ * and is available at http://www.eclipse.org/legal/epl-v20.html
+ *
+ * Contributors:
+ * Red Hat, Inc. - initial API and implementation
+ ******************************************************************************/
+
+package com.redhat.devtools.lsp4ij.features.formatting;
+
+import com.intellij.psi.PsiFile;
+import com.intellij.util.containers.ContainerUtil;
+import com.redhat.devtools.lsp4ij.LSPRequestConstants;
+import com.redhat.devtools.lsp4ij.LanguageServerItem;
+import com.redhat.devtools.lsp4ij.features.AbstractLSPDocumentFeatureSupport;
+import com.redhat.devtools.lsp4ij.internal.CancellationSupport;
+import com.redhat.devtools.lsp4ij.internal.CompletableFutures;
+import org.eclipse.lsp4j.DocumentOnTypeFormattingParams;
+import org.eclipse.lsp4j.TextEdit;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * LSP on-type formatting support which uses:
+ *
+ * <ul>
+ *     <li>LSP 'textDocument/onTypeFormatting' requests</li>
+ * </ul>
+ */
+public class LSPOnTypeFormattingSupport extends AbstractLSPDocumentFeatureSupport<DocumentOnTypeFormattingParams, List<TextEdit>> {
+
+    public LSPOnTypeFormattingSupport(@NotNull PsiFile file) {
+        super(file);
+    }
+
+    public CompletableFuture<List<TextEdit>> onTypeFormatting(DocumentOnTypeFormattingParams params) {
+        return super.getFeatureData(params);
+    }
+
+    @Override
+    protected CompletableFuture<List<TextEdit>> doLoad(DocumentOnTypeFormattingParams params, CancellationSupport cancellationSupport) {
+        PsiFile file = super.getFile();
+        return onTypeFormatting(file, params, cancellationSupport);
+    }
+
+    private static @NotNull CompletableFuture<List<TextEdit>> onTypeFormatting(@NotNull PsiFile file,
+                                                                               @NotNull DocumentOnTypeFormattingParams params,
+                                                                               @NotNull CancellationSupport cancellationSupport) {
+        return getLanguageServers(file,
```

Comment:
Replace with:

```java
getLanguageServers(file,
                f -> f.getServerStatus() == ServerStatus.started &&
                                     f.getOnTypeFormattingFeature().isEnabled(file) &&
                                    f.getOnTypeFormattingFeature().isTriggerCharactersSupported(charedType),
```

Reply:
to know how to implement `isTriggerCharactersSupported`, I suggest you do like completion https://github.com/redhat-developer/lsp4ij/blob/9bb7c8d4b2ba70f6e902d739355fdb4754ade3f7/src/main/java/com/redhat/devtools/lsp4ij/client/features/LSPCompletionFeature.java#L111